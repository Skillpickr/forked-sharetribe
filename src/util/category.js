const Categories = {
  creative: 'creative',
  performance: 'performance',
  knowledge: 'knowledge',
  audioProd: 'audio-prod',
  mediaProd: 'media-prod'
}

const Skills = {
  photographer: 'photographer',
  videographer: 'videographer',
  dj: 'dj',
  musician: 'musician',
  band: 'band',
  makeupArtist: 'makeupArtist',
  dancer: 'dancer',
  miscellaneous: 'miscellaneous'
}
export { Categories, Skills }
