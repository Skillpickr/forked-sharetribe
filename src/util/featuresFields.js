const InputFieldsType = {}

const CheckboxFieldsType = {
  photographerTypeKey: 'photographerType',
  djTypeKey: 'djType',
  musicianTypeKey: 'musicianType',
  musicianGenreKey: 'musicianGenre',
  bandGenreKey: 'bandGenre',
  bandTypeKey: 'bandType'
}

const DropdownFieldsType = {
  musicianSoloKey: 'musicianSoloType',
  ownStudioKey: 'ownStudio',
  soundLightExpKey: 'soundLightExp',
  djGearForPlayingKey: 'djGearForPlaying',
  songRequestKey: 'songRequest'
}

export { InputFieldsType, CheckboxFieldsType, DropdownFieldsType }
