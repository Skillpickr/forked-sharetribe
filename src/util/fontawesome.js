// import the library
import { library, config } from '@fortawesome/fontawesome-svg-core'

// import your icons
import { fas, faCircleQuestion, faMagnifyingGlass } from '@fortawesome/free-solid-svg-icons'

import { fab } from '@fortawesome/free-brands-svg-icons'

config.autoAddCss = false
library.add(
  fas,
  fab,
  faCircleQuestion,
  faMagnifyingGlass
  // more icons go here
)
