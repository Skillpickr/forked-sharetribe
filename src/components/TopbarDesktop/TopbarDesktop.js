import React, { useState, useEffect } from 'react'
import PropTypes from 'prop-types'
import { FormattedMessage, intlShape, injectIntl } from '../../util/reactIntl'
import classNames from 'classnames'
import { ACCOUNT_SETTINGS_PAGES } from '../../routeConfiguration'
import { propTypes } from '../../util/types'
import { compose } from 'redux'
import { connect } from 'react-redux'
import { getMarketplaceEntities } from '../../ducks/marketplaceData.duck'
import { isScrollingDisabled } from '../../ducks/UI.duck'
import {
  Avatar,
  InlineTextButton,
  Logo,
  Menu,
  MenuLabel,
  MenuContent,
  MenuItem,
  NamedLink,
  InboxItem,
  IconSpinner
} from '../../components'
import { TopbarSearchForm } from '../../forms'
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome'
import { txState } from '../../containers/InboxPage/tsState'

import css from './TopbarDesktop.module.css'
import config from '../../config'

const TopbarDesktop = (props) => {
  const {
    className,
    currentUser,
    currentPage,
    rootClassName,
    currentUserHasListings,
    notificationCount,
    intl,
    isAuthenticated,
    onLogout,
    onSearchSubmit,
    initialSearchFormValues
  } = props
  const [mounted, setMounted] = useState(false)

  console.log(currentUserHasListings)

  useEffect(() => {
    setMounted(true)
  }, [])

  const authenticatedOnClientSide = mounted && isAuthenticated
  const isAuthenticatedOrJustHydrated = isAuthenticated || !mounted

  const classes = classNames(rootClassName || css.root, className)

  const search = (
    <TopbarSearchForm
      className={css.searchLink}
      desktopInputRoot={css.topbarSearchWithLeftPadding}
      onSubmit={onSearchSubmit}
      initialValues={initialSearchFormValues}
    />
  )

  const notificationDot = notificationCount > 0 ? <div className={css.notificationDot} /> : null
  // const { tab } = params
  // const isOrders = tab === 'orders'
  // const toTxItem = (tx) => {
  //   const stateData = txState(intl, tx)

  //   // Render InboxItem only if the latest transition of the transaction is handled in the `txState` function.
  //   return stateData ? (
  //     <MenuItem key={tx.id.uuid} className={css.listItem}>
  //       <InboxItem unitType={unitType} tx={tx} intl={intl} stateData={stateData} />
  //     </MenuItem>
  //   ) : null
  // }

  // const noResults =
  //   !fetchInProgress && transactions.length === 0 && !fetchOrdersOrSalesError ? (
  //     <MenuItem key="noResults">
  //       <FormattedMessage id={isOrders ? 'InboxPage.noOrdersFound' : 'InboxPage.noSalesFound'} />
  //     </MenuItem>
  //   ) : null
  // TODO: notification menu needs to be loaded on app init
  const inboxLink = authenticatedOnClientSide ? (
    <div>
      {/* <Menu>
        <MenuLabel className={css.inboxLink}>
          <FontAwesomeIcon icon="fa-solid fa-bell" />
          {notificationDot}
        </MenuLabel>
        <MenuContent className={css.profileMenuContent}>
          {!fetchInProgress ? (
            transactions.map(toTxItem)
          ) : (
            <MenuItem key="isLoading">
              <IconSpinner />
            </MenuItem>
          )}
          {noResults}
        </MenuContent>
      </Menu> */}
      <NamedLink
        className={css.inboxLink}
        name="InboxPage"
        params={{ tab: currentUserHasListings ? 'sales' : 'orders' }}>
        <span className={css.inbox}>
          <FontAwesomeIcon icon="fa-solid fa-bell" />
          {notificationDot}
        </span>
      </NamedLink>
      {currentUserHasListings && (
        <NamedLink className={css.inboxLink} name="ManageListingsPage">
          <span className={css.menuItemBorder} />
          {/* <FormattedMessage id="TopbarDesktop.yourListingsLink" /> */}
          <FontAwesomeIcon icon="fa-solid fa-piggy-bank" />{' '}
        </NamedLink>
      )}
    </div>
  ) : null

  const currentPageClass = (page) => {
    const isAccountSettingsPage = page === 'AccountSettingsPage' && ACCOUNT_SETTINGS_PAGES.includes(currentPage)
    return currentPage === page || isAccountSettingsPage ? css.currentPage : null
  }

  const firstNewListing = currentUserHasListings ? (
    <MenuItem key={null}></MenuItem>
  ) : (
    <MenuItem key="FirstNewListingPage">
      <NamedLink className={classNames(css.yourListingsLink, currentPageClass('NewListingPage'))} name="NewListingPage">
        <span className={css.menuItemBorder} />
        <FormattedMessage id="TopbarDesktop.createListing" />
      </NamedLink>
    </MenuItem>
  )

  const profileMenu = authenticatedOnClientSide ? (
    <Menu useArrow={true}>
      <MenuLabel className={css.profileMenuLabel} isOpenClassName={css.profileMenuIsOpen}>
        <Avatar className={css.avatar} user={currentUser} disableProfileLink />
      </MenuLabel>
      <MenuContent className={css.profileMenuContent}>
        {firstNewListing}
        <MenuItem key="InboxPage">
          <NamedLink
            className={classNames(css.yourListingsLink, currentPageClass('InboxPage'))}
            name="InboxPage"
            params={{ tab: currentUserHasListings ? 'sales' : 'orders' }}>
            <span className={css.menuItemBorder} />
            <FormattedMessage id="TopbarDesktop.inbox" />
            {notificationDot}
          </NamedLink>
        </MenuItem>
        {/* <MenuItem key="ManageListingsPage">
          <NamedLink
            className={classNames(css.yourListingsLink, currentPageClass('ManageListingsPage'))}
            name="ManageListingsPage">
            <span className={css.menuItemBorder} />
            <FormattedMessage id="TopbarDesktop.yourListingsLink" />
          </NamedLink>
        </MenuItem> */}
        <MenuItem key="ProfileSettingsPage">
          <NamedLink
            className={classNames(css.profileSettingsLink, currentPageClass('ProfileSettingsPage'))}
            name="ProfileSettingsPage">
            <span className={css.menuItemBorder} />
            <FormattedMessage id="TopbarDesktop.profileSettingsLink" />
          </NamedLink>
        </MenuItem>
        <MenuItem key="AccountSettingsPage">
          <NamedLink
            className={classNames(css.yourListingsLink, currentPageClass('AccountSettingsPage'))}
            name="AccountSettingsPage">
            <span className={css.menuItemBorder} />
            <FormattedMessage id="TopbarDesktop.accountSettingsLink" />
          </NamedLink>
        </MenuItem>
        <MenuItem key="logout">
          <InlineTextButton rootClassName={css.logoutButton} onClick={onLogout}>
            <span className={css.menuItemBorder} />
            <FormattedMessage id="TopbarDesktop.logout" />
          </InlineTextButton>
        </MenuItem>
      </MenuContent>
    </Menu>
  ) : null

  // const signupLink = isAuthenticatedOrJustHydrated ? null : (
  //   <NamedLink name="SignupPage" className={css.signupLink}>
  //     <span className={css.signup}>
  //       <FormattedMessage id="TopbarDesktop.signup" />
  //     </span>
  //   </NamedLink>
  // )

  const loginLink = isAuthenticatedOrJustHydrated ? null : (
    <NamedLink name="LoginPage" className={css.loginLink}>
      <span className={css.login}>
        <FormattedMessage id="TopbarDesktop.login" />
      </span>
    </NamedLink>
  )

  return (
    <nav className={classes}>
      <NamedLink className={css.logoLink} name="LandingPage">
        <Logo format="desktop" className={css.logo} alt={intl.formatMessage({ id: 'TopbarDesktop.logo' })} />
      </NamedLink>
      {search}
      {/* <NamedLink className={css.createListingLink} name="NewListingPage">
        <span className={css.createListing}>
          <FormattedMessage id="TopbarDesktop.createListing" />
        </span>
      </NamedLink> */}
      {inboxLink}
      {profileMenu}
      {/* {signupLink} */}
      {loginLink}
    </nav>
  )
}

const { bool, func, object, number, string } = PropTypes

TopbarDesktop.defaultProps = {
  rootClassName: null,
  className: null,
  currentUser: null,
  currentPage: null,
  notificationCount: 0,
  initialSearchFormValues: {}
}

TopbarDesktop.propTypes = {
  rootClassName: string,
  className: string,
  currentUserHasListings: bool.isRequired,
  currentUser: propTypes.currentUser,
  currentPage: string,
  isAuthenticated: bool.isRequired,
  onLogout: func.isRequired,
  notificationCount: number,
  onSearchSubmit: func.isRequired,
  initialSearchFormValues: object,
  intl: intlShape.isRequired
}

export default TopbarDesktop
