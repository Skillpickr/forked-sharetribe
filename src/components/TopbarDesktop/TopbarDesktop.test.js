import React from 'react'
import { fakeIntl } from '../../util/test-data'
import { renderDeep } from '../../util/test-helpers'
import TopbarDesktop from './TopbarDesktop'

const noop = () => null

// A hack to get the tooltip random className Uuid to be t00000000-0000-4000-8000-000000000000
jest.mock('crypto', () => ({
  randomBytes: (num) => new Array(num).fill(0),
}));

describe('TopbarDesktop', () => {
  it('data matches snapshot', () => {
    window.google = { maps: {} }
    const topbarProps = {
      isAuthenticated: true,
      currentUserHasListings: true,
      name: 'John Doe',
      onSearchSubmit: noop,
      intl: fakeIntl,
      onLogout: noop
    }
    const tree = renderDeep(<TopbarDesktop {...topbarProps} />)
    delete window.google
    expect(tree).toMatchSnapshot()
  })
})
