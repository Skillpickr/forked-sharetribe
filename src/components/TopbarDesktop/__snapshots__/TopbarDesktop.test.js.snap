// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TopbarDesktop data matches snapshot 1`] = `
<nav
  className="root"
>
  <a
    className="logoLink NamedLink_active"
    data-tip={true}
    href="/"
    onClick={[Function]}
    onMouseOver={[Function]}
    onTouchStart={[Function]}
    style={Object {}}
    title={null}
  >
    <img
      alt="TopbarDesktop.logo"
      className="logo"
      src="logo_transparent_background.png"
    />
  </a>
  <form
    action="/"
    className="searchLink"
    method="post"
    onMouseOver={[Function]}
    onSubmit={[Function]}
    onTouchStart={[Function]}
  >
    <div
      className="root topbarSearchWithLeftPadding"
    >
      <div
        className="desktopIcon"
      >
        <svg
          className="iconSvg"
          height="22"
          viewBox="0 0 21 22"
          width="21"
          xmlns="http://www.w3.org/2000/svg"
        >
          <g
            className="iconSvgGroup"
            fill="none"
            fillRule="evenodd"
            strokeLinecap="round"
            strokeLinejoin="round"
            strokeWidth="2"
            transform="matrix(-1 0 0 1 20 1)"
          >
            <path
              d="M13 14l5.241 5.241"
            />
            <circle
              cx="7.5"
              cy="7.5"
              r="7.5"
            />
          </g>
        </svg>
      </div>
      <input
        autoComplete="off"
        autoFocus={false}
        className="desktopInput"
        disabled={false}
        name="location"
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        placeholder="TopbarSearchForm.placeholder"
        type="search"
        value=""
      />
    </div>
  </form>
  <div>
    <div>
      <a
        className="inboxLink"
        data-for="yourListingsLink"
        data-tip={true}
        href="/listings"
        onClick={[Function]}
        onMouseOver={[Function]}
        onTouchStart={[Function]}
        style={Object {}}
        title={null}
      >
        <span
          className="inbox"
        />
      </a>
      <div
        className="__react_component_tooltip t98a893a9-a2c3-4da7-b930-8dc00331050d place-top type-dark"
        data-id="tooltip"
        id="yourListingsLink"
      >
        <style
          aria-hidden="true"
          dangerouslySetInnerHTML={
            Object {
              "__html": "
  	.t98a893a9-a2c3-4da7-b930-8dc00331050d {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
	    border-radius: undefinedpx;
	    padding: 8px 21px;
  	}

  	.t98a893a9-a2c3-4da7-b930-8dc00331050d.place-top {
        margin-top: -10px;
    }
    .t98a893a9-a2c3-4da7-b930-8dc00331050d.place-top::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: 2;
        width: 20px;
        height: 12px;
    }
    .t98a893a9-a2c3-4da7-b930-8dc00331050d.place-top::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        bottom: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(135deg);
    }

    .t98a893a9-a2c3-4da7-b930-8dc00331050d.place-bottom {
        margin-top: 10px;
    }
    .t98a893a9-a2c3-4da7-b930-8dc00331050d.place-bottom::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 18px;
        height: 10px;
    }
    .t98a893a9-a2c3-4da7-b930-8dc00331050d.place-bottom::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        top: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(45deg);
    }

    .t98a893a9-a2c3-4da7-b930-8dc00331050d.place-left {
        margin-left: -10px;
    }
    .t98a893a9-a2c3-4da7-b930-8dc00331050d.place-left::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .t98a893a9-a2c3-4da7-b930-8dc00331050d.place-left::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        right: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(45deg);
    }

    .t98a893a9-a2c3-4da7-b930-8dc00331050d.place-right {
        margin-left: 10px;
    }
    .t98a893a9-a2c3-4da7-b930-8dc00331050d.place-right::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .t98a893a9-a2c3-4da7-b930-8dc00331050d.place-right::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        left: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(-135deg);
    }
  ",
            }
          }
        />
        <span>
          <span>
            TopbarDesktop.yourListingsLink
          </span>
        </span>
      </div>
    </div>
  </div>
  <div
    className="__react_component_tooltip t63c11e09-1480-46e6-947c-30a80556a0be place-top type-dark"
    data-id="tooltip"
    id="yourSwitchLanguageLink"
  >
    <style
      aria-hidden="true"
      dangerouslySetInnerHTML={
        Object {
          "__html": "
  	.t63c11e09-1480-46e6-947c-30a80556a0be {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
	    border-radius: undefinedpx;
	    padding: 8px 21px;
  	}

  	.t63c11e09-1480-46e6-947c-30a80556a0be.place-top {
        margin-top: -10px;
    }
    .t63c11e09-1480-46e6-947c-30a80556a0be.place-top::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: 2;
        width: 20px;
        height: 12px;
    }
    .t63c11e09-1480-46e6-947c-30a80556a0be.place-top::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        bottom: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(135deg);
    }

    .t63c11e09-1480-46e6-947c-30a80556a0be.place-bottom {
        margin-top: 10px;
    }
    .t63c11e09-1480-46e6-947c-30a80556a0be.place-bottom::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 18px;
        height: 10px;
    }
    .t63c11e09-1480-46e6-947c-30a80556a0be.place-bottom::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        top: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(45deg);
    }

    .t63c11e09-1480-46e6-947c-30a80556a0be.place-left {
        margin-left: -10px;
    }
    .t63c11e09-1480-46e6-947c-30a80556a0be.place-left::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .t63c11e09-1480-46e6-947c-30a80556a0be.place-left::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        right: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(45deg);
    }

    .t63c11e09-1480-46e6-947c-30a80556a0be.place-right {
        margin-left: 10px;
    }
    .t63c11e09-1480-46e6-947c-30a80556a0be.place-right::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .t63c11e09-1480-46e6-947c-30a80556a0be.place-right::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        left: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(-135deg);
    }
  ",
        }
      }
    />
    <span>
      <span>
        TopbarDesktop.yourSwitchLanguageLink
      </span>
    </span>
  </div>
  <a
    className="loginLink"
    data-for="yourSwitchLanguageLinkContainer"
    data-tip={true}
  >
    <span
      className="inbox"
    />
  </a>
  <div
    className="__react_component_tooltip tf0b92bc8-94b5-42f4-a388-f99f38442e7f place-top type-dark"
    data-id="tooltip"
    id="yourSwitchLanguageLinkContainer"
  >
    <style
      aria-hidden="true"
      dangerouslySetInnerHTML={
        Object {
          "__html": "
  	.tf0b92bc8-94b5-42f4-a388-f99f38442e7f {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
	    border-radius: undefinedpx;
	    padding: 8px 21px;
  	}

  	.tf0b92bc8-94b5-42f4-a388-f99f38442e7f.place-top {
        margin-top: -10px;
    }
    .tf0b92bc8-94b5-42f4-a388-f99f38442e7f.place-top::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: 2;
        width: 20px;
        height: 12px;
    }
    .tf0b92bc8-94b5-42f4-a388-f99f38442e7f.place-top::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        bottom: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(135deg);
    }

    .tf0b92bc8-94b5-42f4-a388-f99f38442e7f.place-bottom {
        margin-top: 10px;
    }
    .tf0b92bc8-94b5-42f4-a388-f99f38442e7f.place-bottom::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 18px;
        height: 10px;
    }
    .tf0b92bc8-94b5-42f4-a388-f99f38442e7f.place-bottom::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        top: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(45deg);
    }

    .tf0b92bc8-94b5-42f4-a388-f99f38442e7f.place-left {
        margin-left: -10px;
    }
    .tf0b92bc8-94b5-42f4-a388-f99f38442e7f.place-left::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .tf0b92bc8-94b5-42f4-a388-f99f38442e7f.place-left::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        right: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(45deg);
    }

    .tf0b92bc8-94b5-42f4-a388-f99f38442e7f.place-right {
        margin-left: 10px;
    }
    .tf0b92bc8-94b5-42f4-a388-f99f38442e7f.place-right::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .tf0b92bc8-94b5-42f4-a388-f99f38442e7f.place-right::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        left: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(-135deg);
    }
  ",
        }
      }
    />
    <span>
      <span>
        TopbarDesktop.yourSwitchLanguageLink
      </span>
    </span>
  </div>
  <div>
    <div
      className="root"
      data-tip={true}
      onBlur={[Function]}
      onKeyDown={[Function]}
      tabIndex={0}
    >
      <button
        className="root profileMenuLabel"
        data-for="yourProfileLinkContainer"
        data-tip={true}
        onBlur={[Function]}
        onClick={[Function]}
      >
        <div
          className="root avatar"
          title=""
        >
          <span
            className="initials"
          />
        </div>
      </button>
      <div
        className="__react_component_tooltip t9d9e18b1-c38a-4246-b799-336fc6a0a787 place-top type-dark"
        data-id="tooltip"
        id="yourProfileLinkContainer"
      >
        <style
          aria-hidden="true"
          dangerouslySetInnerHTML={
            Object {
              "__html": "
  	.t9d9e18b1-c38a-4246-b799-336fc6a0a787 {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
	    border-radius: undefinedpx;
	    padding: 8px 21px;
  	}

  	.t9d9e18b1-c38a-4246-b799-336fc6a0a787.place-top {
        margin-top: -10px;
    }
    .t9d9e18b1-c38a-4246-b799-336fc6a0a787.place-top::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: 2;
        width: 20px;
        height: 12px;
    }
    .t9d9e18b1-c38a-4246-b799-336fc6a0a787.place-top::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        bottom: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(135deg);
    }

    .t9d9e18b1-c38a-4246-b799-336fc6a0a787.place-bottom {
        margin-top: 10px;
    }
    .t9d9e18b1-c38a-4246-b799-336fc6a0a787.place-bottom::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 18px;
        height: 10px;
    }
    .t9d9e18b1-c38a-4246-b799-336fc6a0a787.place-bottom::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        top: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(45deg);
    }

    .t9d9e18b1-c38a-4246-b799-336fc6a0a787.place-left {
        margin-left: -10px;
    }
    .t9d9e18b1-c38a-4246-b799-336fc6a0a787.place-left::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .t9d9e18b1-c38a-4246-b799-336fc6a0a787.place-left::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        right: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(45deg);
    }

    .t9d9e18b1-c38a-4246-b799-336fc6a0a787.place-right {
        margin-left: 10px;
    }
    .t9d9e18b1-c38a-4246-b799-336fc6a0a787.place-right::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .t9d9e18b1-c38a-4246-b799-336fc6a0a787.place-right::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        left: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(-135deg);
    }
  ",
            }
          }
        />
        <span>
          <span>
            TopbarDesktop.yourProfileLink
          </span>
        </span>
      </div>
      <div
        className="root profileMenuContent isClosed"
        style={
          Object {
            "left": "-10000px",
          }
        }
      >
        <ul
          className="content"
        >
          <li
            className="root"
            role="menuitem"
          />
          <li
            className="root"
            role="menuitem"
          >
            <a
              className="yourListingsLink"
              data-tip={true}
              href="/inbox/sales"
              onClick={[Function]}
              onMouseOver={[Function]}
              onTouchStart={[Function]}
              style={Object {}}
              title={null}
            >
              <span
                className="menuItemBorder"
              />
              <span>
                TopbarDesktop.inbox
              </span>
            </a>
          </li>
          <li
            className="root"
            role="menuitem"
          >
            <a
              className="profileSettingsLink"
              data-tip={true}
              href="/profile-settings"
              onClick={[Function]}
              onMouseOver={[Function]}
              onTouchStart={[Function]}
              style={Object {}}
              title={null}
            >
              <span
                className="menuItemBorder"
              />
              <span>
                TopbarDesktop.profileSettingsLink
              </span>
            </a>
          </li>
          <li
            className="root"
            role="menuitem"
          >
            <a
              className="yourListingsLink"
              data-tip={true}
              href="/account"
              onClick={[Function]}
              onMouseOver={[Function]}
              onTouchStart={[Function]}
              style={Object {}}
              title={null}
            >
              <span
                className="menuItemBorder"
              />
              <span>
                TopbarDesktop.accountSettingsLink
              </span>
            </a>
          </li>
          <li
            className="root"
            role="menuitem"
          >
            <button
              className="logoutButton inlineTextButton"
              disabled={false}
              onClick={[Function]}
            >
              <span
                className="menuItemBorder"
              />
              <span>
                TopbarDesktop.logout
              </span>
            </button>
          </li>
        </ul>
      </div>
    </div>
  </div>
  <div>
    <a
      className="inboxLink"
      data-for="notificationTooltip"
      data-tip={true}
      href="/inbox/sales"
      onClick={[Function]}
      onMouseOver={[Function]}
      onTouchStart={[Function]}
      style={Object {}}
      title={null}
    >
      <span
        className="inbox"
      />
    </a>
    <div
      className="__react_component_tooltip t50c5d4cc-a0de-41a9-b751-5ae20921831a place-top type-dark"
      data-id="tooltip"
      id="notificationTooltip"
    >
      <style
        aria-hidden="true"
        dangerouslySetInnerHTML={
          Object {
            "__html": "
  	.t50c5d4cc-a0de-41a9-b751-5ae20921831a {
	    color: #fff;
	    background: #222;
	    border: 1px solid transparent;
	    border-radius: undefinedpx;
	    padding: 8px 21px;
  	}

  	.t50c5d4cc-a0de-41a9-b751-5ae20921831a.place-top {
        margin-top: -10px;
    }
    .t50c5d4cc-a0de-41a9-b751-5ae20921831a.place-top::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: 2;
        width: 20px;
        height: 12px;
    }
    .t50c5d4cc-a0de-41a9-b751-5ae20921831a.place-top::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        bottom: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(135deg);
    }

    .t50c5d4cc-a0de-41a9-b751-5ae20921831a.place-bottom {
        margin-top: 10px;
    }
    .t50c5d4cc-a0de-41a9-b751-5ae20921831a.place-bottom::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 18px;
        height: 10px;
    }
    .t50c5d4cc-a0de-41a9-b751-5ae20921831a.place-bottom::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        top: -6px;
        left: 50%;
        margin-left: -6px;
        transform: rotate(45deg);
    }

    .t50c5d4cc-a0de-41a9-b751-5ae20921831a.place-left {
        margin-left: -10px;
    }
    .t50c5d4cc-a0de-41a9-b751-5ae20921831a.place-left::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .t50c5d4cc-a0de-41a9-b751-5ae20921831a.place-left::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        right: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(45deg);
    }

    .t50c5d4cc-a0de-41a9-b751-5ae20921831a.place-right {
        margin-left: 10px;
    }
    .t50c5d4cc-a0de-41a9-b751-5ae20921831a.place-right::before {
        content: \\"\\";
        background-color: inherit;
        position: absolute;
        z-index: -1;
        width: 10px;
        height: 18px;
    }
    .t50c5d4cc-a0de-41a9-b751-5ae20921831a.place-right::after {
        content: \\"\\";
        position: absolute;
        width: 10px;
        height: 10px;
        border-top-right-radius: undefinedpx;
        border: 1px solid transparent;
        background-color: #222;
        z-index: -2;
        left: -6px;
        top: 50%;
        margin-top: -6px;
        transform: rotate(-135deg);
    }
  ",
          }
        }
      />
      <span>
        <span>
          TopbarDesktop.notification
        </span>
      </span>
    </div>
  </div>
</nav>
`;
